torture/src/supervisor/mod.rs

```diff
@@ -212,26 +217,43 @@ async fn control_loop(
     workload_params: WorkloadParams,
     flag_num_limit: usize,
 ) -> Result<()> {
+    info!("Starting control loop, seed={seed}.\n{NON_DETERMINISM_DISCLAIMER}");
     let mut flags = Vec::new();
     let mut workload_cnt = 0;
-    // TODO: Run workloads in parallel. Make the concurrency factor configurable.
-    info!("Starting control loop, seed={seed}.\n{NON_DETERMINISM_DISCLAIMER}");
+    const MAX_PARALLEL_WORKLOADS: usize = 1;
+    let mut running_workloads = JoinSet::new();
+
     loop {
-        let workload_id = workload_cnt;
-        let workload_seed = seed + workload_cnt;
-        workload_cnt += 1;
-        let maybe_flag = run_workload(
-            cancel_token.clone(),
-            workload_seed,
-            &workload_params,
-            workload_id,
-        )
-        .instrument(trace_span!("workload", workload_id))
-        .await?;
+        // Collect any finished workload.
+        while let Some(maybe_flag) = running_workloads.try_join_next() {
+            if let Some(flag) = maybe_flag?? {
+                print_flag(&flag);
+                flags.push(flag);
+            }
+        }
+
+        while running_workloads.len() < MAX_PARALLEL_WORKLOADS {
+            let workload_id = workload_cnt;
+            let workload_seed = seed + workload_cnt;
+            workload_cnt += 1;
+
+            let _ = running_workloads.spawn(run_workload(
+                cancel_token.clone(),
+                workload_seed,
+                workload_params.clone(),
+                workload_id,
+            ));
+        }
+
+        // Here is safe to wait on a workload completion because if
+        // ctrl-c is received, the cancel_token will stop the workload
+        // and thus allow all workloads to conclude early.
+        let maybe_flag = running_workloads.join_next().await.unwrap()??;
```

Comment:
Nit: unwrap comment.

Plus: this is pretty piled up: unwrap() + ? + ?, not clear what is what, and ideally we spread it out, annotate types, and/or document.